### ISSUE TITLE
Collapsed items, not filling the whole scrolling space

### ISSUE THREAD

**Issue by graphefruit** [ericferreira1992/ag-virtual-scroll#28]
Hey @ericferreira1992,

I've added in a last update in my app, the possibility to "collapse" the coffee brews.
Interesting is that I got about 789 items, and 4 are just rendered, even so there would be enough space for 2 more:
<img width="1332" alt="image" src="https://github.com/user-attachments/assets/4ad89484-96d7-4c8b-bb0f-de0db3233d31">

When I expand the brews it takes the whole scroll container:
<img width="1312" alt="image" src="https://github.com/user-attachments/assets/d5efaf08-4738-4c9a-acb2-f539ffe5b1be">

Do you have any idea whats happening here and what I could do?

My container is set like:
```
 <ag-virtual-scroll #archivedScroll [items]="archiveBrewsView" height="100vh"  [min-row-height]="isCollapseActive()?98:180">
 ```
 Setting the min-row-height to "auto" is a bit fixing this issue, but still not taking up the whole space.
 
 Thank you so much for your support!

**Comment by graphefruit** [comment]
Adding to this.

The issue coming from was the attribute `height=100vh`

The solution I did now was:
``` [height]="segmentScrollHeight" ```

and in code behind I calculate and set this:
 ``` this.segmentScrollHeight = calculcatedPX;```
 
 And then I recalculate aswell when `orientation-change` is triggered.
 
 The funny thing is rather: 100vh is working on expanded elements but not on collapsed once

**Comment by ericferreira1992** [comment]
@graphefruit, I just launched a new v1.9.2, fixing this bug in ag-virtual-scroll. Try to update and let me know after.

**Comment by graphefruit** [comment]
Hey @ericferreira1992 
Thanks for the fix.
Yes and no in once.

I've changed my code back to 100vh, and now it collapse to the bottom, but the difference between 1.9.0 and 1.9.2, is that there is now very much blank space:

1.9.0:
<img width="363" alt="image" src="https://github.com/user-attachments/assets/332a21ca-d5cf-4930-a3ad-5b4fd2d1248b">

1.9.2:
<img width="364" alt="image" src="https://github.com/user-attachments/assets/b33264b6-a0e7-40b5-9aba-bb18356d102c">

on bigger tiles this issue shines more:
<img width="355" alt="image" src="https://github.com/user-attachments/assets/f1310bbe-8ee3-4159-ab5d-1fc870a3e565">

**Comment by ericferreira1992** [comment]
I'm struggling to replicate the error 😅. But I'll get it done and solve it! I let you know @graphefruit.

**Comment by graphefruit** [comment]
@ericferreira1992 
If you're a bit familiar with Ionic you can install the develop-branch of Beanconqueror.
I can also give you a helping hand so.
Thanks for your support!

**Comment by graphefruit** [comment]
Hey @ericferreira1992,
coming back to this case.
I've strangely encountered one strange new case:
In the first comment I've added that when I calculate the pixel-size for the height its working.
I've now encountered one specific situation where its not working:
<img width="1351" alt="image" src="https://github.com/user-attachments/assets/67fd0800-a343-4c7b-83c9-da4d997457f2">

The size is set and there are just 5 elements are rendered.
BUT I've encountered a realy huge lag when opening the screen, and found out, that all 240 entries (which are in the list) are initialized, even when there are just 5 visible / rendered:
<img width="1054" alt="image" src="https://github.com/user-attachments/assets/2193768b-8796-4448-a699-3d46a5ee03a0">


on the other solutions on the other pages where its identically implemented, are just initializing the needed brews.


And now the funfact: 
```
 <ag-virtual-scroll #openScrollAssociatedBrews [items]="associatedBrews" [height]="segmentScrollHeight" min-row-height="180">
 ```
If I change `segmentScrollHeight` to -> `'100vh' -> its working.

Maybe this helps for some more debugging

**Comment by graphefruit** [comment]
I'd also have the TRACE, but its to big, but maybe the screenshot helps you aswell:
![image](https://github.com/user-attachments/assets/afc80672-c4fc-4109-9a1a-8ce00122b284)

If needed I can pass you the trace as a link via email

**Comment by graphefruit** [comment]
@ericferreira1992 do you run it on browser? 

I can checkin you some sample data, but you'd also need to revert from 100vh to my calculcations e.g.

**Comment by ericferreira1992** [comment]
Yes, I'm using browser (G. Chrome). Ok, I'm pointing to the loca `ag-virtual-scroll` source code to debug that situation.

**Comment by graphefruit** [comment]
Good evening @ericferreira1992 
First of all: Thank you for your time and effort chasing this situation.
I've pushed to develop the sample-data.
Attached a short video what you need to do and to see this issue:

https://github.com/user-attachments/assets/37247848-51cd-40e9-bccc-186997a5427c

I've also opened up one ticket, where I put the code change into it, to see where you can start to debug / change to see what is the differene: https://github.com/graphefruit/Beanconqueror/issues/838

If you need anything else I'm happy to help!
Lars

**Comment by ericferreira1992** [comment]
Right! I ran the application here. I noticed the issue is because how your brews page is rendered at the application. You will never be able set `height="100%'"` or `height="100vh"` and get success.
- `height="100%'"`: the parent elements aren't stretched (with `height="100%"` too) enough so `ag-virtual-scroll` can strech too.
- `height="100vh"`: this might work, but always will render a `ag-virtual-scroll` with a big height (same height as the screen) causing external scrolls, so you will have two scrolls, one from do app (in a specific ion parent element) and another from `ag-virtual-scroll` component. In this case you will always need to set calculated height like `height="calc(100vh - 66px)"`, which `66px` would be the `ion-segment` rendered above.

I managed to make `ag-virtual-scroll` in this page with lastest version perfectly, calculating height as I said above. Try do this and tell me more.

**Comment by graphefruit** [comment]
Thank you for the digging::
> You will never be able set `height="100%'"` or `height="100vh"` and get success.

Somehow the funny thing is: This worked very long 

>In this case you will always need to set calculated height like height="calc(100vh - 66px)", which 66px would be the ion-segment rendered above.

This is what I'm doing in the associcated brews, I'll update to the latest version (I just calculate this in a variable and set it) 
https://github.com/graphefruit/Beanconqueror/commit/4d2f01a01494f5bec56186d62ba503f850b0f10d

I'll try it this evening and give you feedback.

Thanks for your support!
Lars

**Comment by ericferreira1992** [comment]
@graphefruit, I made mistake in my last comment. The calc is wrong, if you use 100vh you going to need to subtract the all elements heights which are taking up space, i.e. `height="calc(100vh - ionSegment - tabBar - navBar)" `. I also did a test stretching parents elements and I managed to use `height="100%”` in `ag-virtual-scroll` dynamically.

**Comment by graphefruit** [comment]
> @graphefruit, I made a mistake in my last comment. The calc is wrong, if you use 100vh you going to need to subtract the all elements heights which are taking up space, i.e. `height="calc(100vh - ionSegment - tabBar - navBar)" `. I also did a test stretching parents elements and I managed to use `height="100%”` in `ag-virtual-scroll` dynamically.

This is what I'M doing in the associated-brew already
```
rivate retriggerScroll() {
    setTimeout(async () => {
      const el = this.associatedBrewsComponent.nativeElement;
      const scrollComponent: AgVirtualSrollComponent =
        this.openScrollAssociatedBrews;

      if (scrollComponent) {
        scrollComponent.el.style.height =
          el.offsetHeight - scrollComponent.el.offsetTop - 20 + 'px';

        this.segmentScrollHeight = scrollComponent.el.style.height;
        setTimeout(() => {
          /** If we wouldn't do it, and the tiles are collapsed, the next once just exist when the user starts scrolling**/
          const elScroll = scrollComponent.el;
          elScroll.dispatchEvent(new Event('scroll'));
        }, 15);
      }
    }, 150);
  }

```

And then I set the height element: `[height]='segmentScrollHeight'`

Btw: I need to do the event-scroll, because if i collapsed the items, sometimes they are not rendered to the bottom - maybe even a side-effect to this case here.

**Comment by graphefruit** [comment]
I've found another very strange situation.

https://github.com/user-attachments/assets/293d7cc3-c8ae-400e-ae9a-a33a2a967a59

When using the PX-Calculcation with a value, and switching from the tabs to another menu and coming back, the whole app freezes.

Intersting:
When I go from: `[height]="segmentScrollHeight"` to: `[height]="'100%'"` the app does not freeze anymore :O

**Comment by graphefruit** [comment]
Second one:
I've just updated to 1.9.2 - with height=100%, it doesn't work anymore at all, when I now add:
```
   const elScroll = scrollComponent.el;
   elScroll.dispatchEvent(new Event('scroll'));
```
As a trigger, it works.

This code I've found as a work around, because the tiles where sometimes not rendered when collapsed (and just 80px height, instead of 180) and therefore displaying much more then before
